### PR TITLE
Issue-Fix-#2755 chore(test): replaced mocks with concrete implementation

### LINKED ISSUE #2755: ImageConfigurationTest : Remove usage of `org.mockito.Mockito.mock` with real objects using builders
### Component

JKube Kit

### Task description

Related to https://github.com/eclipse/jkube/issues/2316

ImageConfigurationTest seems to be using mock here:

https://github.com/eclipse/jkube/blob/3afc7bf1f8dbe4452826c5ec7ec58de5c9408d53/jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageConfigurationTest.java#L30-L31

This could easily be refactored using lombok builders like this:
```java
        // Given + When
        final ImageConfiguration result = ImageConfiguration.builder()
                .name("1337")
                .build(BuildConfiguration.builder()
                    .user("super-user")
                    .build())
                .build();
```

### Expected Behavior
Usage of org.mockito.Mockito.mock is removed from ImageConfigurationTest

### Acceptance Criteria
- [x]  mock statements are replaced via real variables using lombok builders
- [x]  ImageConfigurationTest should be passing even after refactoring the test

#### Kubernetes

If you don't have a real Kubernetes cluster available (most probably), you can use Minikube or Kind to test with a local cluster.

- Minikube [get started guide](https://minikube.sigs.k8s.io/docs/start/)
- Kind (Requires Docker) [quick start](https://kind.sigs.k8s.io/docs/user/quick-start/)

#### OpenShift

If you don't have a real OpenShift cluster available (most probably), you can use [Red Hat's developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox). The only requirement is to have a Red Hat account.

Once you have your Sandbox environment, you'll need to download the oc tool from the cluster console.
(Press the `?` icon and from the context menu select `Command line tools`, you'll be redirected to https://$subdomain.openshiftapps.com/command-lines-tools where you'll be able to download the CLI for your platform)

### PR DESCRIPTION
**Pull request by l3002**
## Description

Fixes #2755 

Replaced mock with concrete implementation of `BuildConfiguration` in `ImageConfigurationTest`

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2763 (_2024-03-01T12:36:56Z_)

:gear: [JKube E2E Tests (8111447728)](https://github.com/jkubeio/ci/actions/runs/8111447728) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2763)](https://github.com/jkubeio/ci/actions/runs/8111447728/job/22170726258)

File: jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageConfigurationTest.java [hunk 1]
```diff
@@ -27,12 +27,13 @@ class ImageConfigurationTest {
     @Test
     void testBuilder() {
         // Given
-        BuildConfiguration mockJKubeBuildConfiguration = mock(BuildConfiguration.class);
-        when(mockJKubeBuildConfiguration.getUser()).thenReturn("super-user");
```
Comment (rohanKanojia) [inline jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageConfigurationTest.java:31]: Could you please remove import statements for mockito now that they are not used?

Comment (l3002) [reply jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageConfigurationTest.java:31]: Just pushed a commit of that, review it once you get time.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**